File: lib/java/src/main/java/io/github/brenbar/uapi/internal/_SchemaParseFailure.java
package io.github.brenbar.uapi.internal;

import java.util.List;
import java.util.Map;

public class _SchemaParseFailure {
    public final List<Object> path;
    public final String reason;
    public final Map<String, Object> data;
    public final String key;

    public _SchemaParseFailure(List<Object> path, String reason, Map<String, Object> data, String key) {
        this.path = path;
        this.reason = reason;
        this.data = data;
        this.key = key;
    }
}

File: lib/ts/uapi/internal/schema/SchemaParseFailure.ts
export class SchemaParseFailure {
    path: any[];
    reason: string;
    data: Record<string, any>;
    key: string | null;

    constructor(path: any[], reason: string, data: Record<string, any>, key: string | null) {
        this.path = path;
        this.reason = reason;
        this.data = data;
        this.key = key;
    }
}
